### ISSUE TITLE
Import Keyword list as Bookmarks

### ISSUE THREAD

**Issue by gfd2020** [sepinf-inc/IPED#1768]
Some users asked me to implement this functionality below. Basically, it's the possibility to create a bookmark for the imported list and/or a bookmark for each word in the imported list. Is there interest in this implementation? It will be very easy.

![ko](https://github.com/sepinf-inc/IPED/assets/59742865/05f05383-6405-4c53-8ce0-088c456047d8)

**Comment by lfcnassif** [comment]
Thanks @gfd2020, it was already asked by our users too, it would be useful. But there is an important limitation: bookmarks would not highlight file names or content in the Preview tab. For this feature, I think we should extend the Bookmark class (by a subclass) to keep track of the original keyword term, so it could be used to highlight file names and content.

**Comment by lfcnassif** [comment]
Maybe we wouldn't need to subclass Bookmarks class, but just use a generic internal Map I added in version 4.0.0 to handle this kind of extension:
https://github.com/sepinf-inc/IPED/blob/5e59ddea31c8b37c76c1906aeef753cb4cbc71de/iped-engine/src/main/java/iped/engine/data/Bookmarks.java#L72

**Comment by gfd2020** [comment]
> Thanks @gfd2020, it was already asked by our users too, it would be very useful. But there is an important limitation: bookmarks would not highlight hits in file names or content in the Preview tab when clicking on contained items.
> 
> For this feature, I think we should extend the Bookmark class (by a subclass) to keep track of the original keyword term, so it could be used to highlight hits in file names and content.

Do you think this is expected behavior?

I made a PR. #1771. No highlighs now ... needs to be implemented.

**Comment by lfcnassif** [comment]
> Do you think this is expected behavior?

No highlights is the Bookmarks expected behavior today. But my concern about this feature is encouraging users to click on Bookmarks instead of on Keywords List Combobox in the search bar and miss the very useful highlights, which are very important for users in my opinion.

**Comment by gfd2020** [comment]
> > Do you think this is expected behavior?
> 
> No highlights is the Bookmarks expected behavior today. But my concern about this feature is encouraging users to click on Bookmarks instead of on Keywords List Combobox in the search bar and miss the very useful highlights, which are very important for users in my opinion.

I see. I think it's a difficult decision.

**Comment by hilderonny** [comment]
Is there any progress to this functionality? Our users also requested such a behaviour. They have a keyword list for international terrorism containing over 600 words in different languages and idioms. Importing this list as keyword list is not elegant enough because you need to perform a separate search for every term and create a bookmark for it manually.

Especially for creating HTML reports it would be very comfortable to create a report by selecting all of the imported search-term-bookmarks and see them in the output.

**Comment by lfcnassif** [comment]
Hi @hilderonny and @gfd2020! I'll try to review #1771 when I return from vacation next week. In my opinion, it would be important to implement keywords highlighting when clicking on bookmarks created from keywords, as I explained above, but maybe that can be left as a future improvement.

**Comment by lfcnassif** [comment]
Hi @gfd2020. I just reviewed #1771 and it worked fine! I just changed a bit the FileChooser layout to show the new checkboxes in a vertical layout, I think users will notice the new options easier, adjusted localization strings to be more intuitive and added a bookmark name prefix and comments:

![image](https://github.com/sepinf-inc/IPED/assets/7276994/222b1c77-78e5-4357-936b-739a1952ef74)

![image](https://github.com/sepinf-inc/IPED/assets/7276994/20e8131e-079e-4a93-af2f-5b1d883a6be9)

I'll leave hit highlighting when filtering by the new keyword bookmarks as a future improvement.
